packages/graphql-migrations/src/migrations/KnexMigrationManager.ts

```diff
@@ -0,0 +1,196 @@
+          if (method) {
+            table[method](field.name);
+          }
+        }
+        table.timestamps();
+      }
+    ).toQuery();
+
+    return `${sqlStatement};`;
+  }
+
+  /**
+   * Generates a SQL statement for altering a table
+   *
+   * @param {InputModelTypeContext}
+   * @param {ModelChange[]} changes
+   * @returns {string}
+   * @memberof KnexMigrationManager
+   */
+  public alterTable(t: InputModelTypeContext, changes: ModelChange[]): string {
+    const tableName = t.name.toLowerCase();
+    const typeChanges = changes.filter((c: ModelChange) => c.path.type === t.name);
+
+    const sqlStatement = this.db.schema.alterTable(
+      tableName,
+      (table: knex.TableBuilder) => {
+        for (const change of typeChanges) {
+          const field = t.fields.find((f: InputModelFieldContext) => f.name === change.path.field);
+          const method = this.primitiveTypesMapping[field.type];
+          if (method) {
+            table[method](field.name);
+          }
+        }
+      }
+    ).toQuery();
+
+    return `${sqlStatement};`;
+  }
+
+  /**
+   * Saves a migration to the database
+   *
+   * @param {SchemaMigration} migration
+   * @returns {Promise<void>}
+   * @memberof KnexMigrationManager
+   */
+  public async createMigration(migration: SchemaMigration): Promise<void> {
+    try {
+      // tslint:disable-next-line: await-promise
+      await this.db.table(this.tables.migrations).insert(migration);
+    } catch (err) {
+      handleError(err);
+    }
+
+    return Promise.resolve();
+  }
+
+  /**
+   * Create migration metadata table(s)
+   *
+   * @returns {Promise<void>}
+   * @memberof KnexMigrationManager
+   */
+  public async createMetadataTables(): Promise<void> {
+    const hasTable = await this.db.schema.hasTable(this.tables.migrations);
+
+    if (!hasTable) {
+      try {
+        // tslint:disable-next-line: await-promise
+        await this.db.schema.createTable(this.tables.migrations, (table: knex.TableBuilder) => {
+          table.bigInteger('id').primary();
+          table.timestamp('applied_at').nullable();
+          table.text('model').notNullable();
+          table.json('changes').nullable();
+          table.text('sql_up').nullable();
+          table.timestamp('rollback_at').nullable();
+        });
+
+        logInfo('Created migrations table');
+      } catch (err) {
+        logError(err);
+      }
+
+    }
+
+    return Promise.resolve();
+  }
+
+  /**
+   * Applies a migration to the database
+   *
+   * @param {SchemaMigration} migration
+   * @returns {Promise<void>}
+   * @memberof KnexMigrationManager
+   */
+  public async applyMigration(migration: SchemaMigration): Promise<void> {
+    try {
+      // tslint:disable-next-line: await-promise
+      await this.db.raw(migration.sql_up);
+
+      logInfo(`Executed migration ${chalk.cyan(migration.id.toString())}`)
+    } catch (err) {
+      handleError(err);
+    }
+
+    const now = new Date();
```

Comment:
As we do not have rollback this can leave entire db in unfinished state meaning that there will be no way to revert or recover other than just wiping the database